### PR TITLE
Simplified standardization

### PR DESCRIPTION
**Pull request by rspeer**
Simplify the tokenization process, and move all standardization-related operations into `conceptnet5.nodes` instead of having them messily sprawled between `uri` and `nodes`.

ConceptNet no longer uses NLTK's `punkt` tokenizer; it instead uses the one from wordfreq 1.0, which splits on spaces and punctuation.

### CODE REVIEW

File: conceptnet5/language/english.py [hunk 1]
```diff
@@ -161,55 +161,12 @@ def morphy_stem(word, pos=None):
...
-    Returns (name, disambig), where "name" is the standardized topic name,
-    and "disambig" is a string corresponding to the disambiguation text or
-    None.
-    """
-    # find titles of the form Foo (bar)
-    topic = topic.replace('_', ' ')
-    match = re.match(r'([^(]+) \(([^)]+)\)', topic)
-    if not match:
-        return standardize(topic), None
+def english_filter(tokens):
```
Comment (Joshua-Chin) [inline conceptnet5/language/english.py]: Can this function get a bit of documentation?

File: conceptnet5/nodes.py [hunk 2]
```diff
@@ -38,6 +41,86 @@
 }
 
 
+def standardize_as_list(text, token_filter=None):
```
Comment (Joshua-Chin) [inline conceptnet5/nodes.py]: token_filter is undocumented. The documentation seems to suggest that token_filter should instead be a flag to apply english_filter.

File: conceptnet5/nodes.py [hunk 3]
```diff
@@ -50,14 +133,16 @@ def standardized_concept_name(lang, text):
...
     'esto_es_una_prueba'
     """
+    lang_filter = None
     if lang == 'en':
-        stem = standardize_english(text) or text
-        return standardize_text(stem)
+        lang_filter = english_filter
     else:
-        return standardize_text(text)
+        lang_filter = None
```
Comment (Joshua-Chin) [inline conceptnet5/nodes.py]: This else clause is unnecessary.

File: conceptnet5/readers/conceptnet4.py [hunk 4]
```diff
@@ -85,9 +87,9 @@ def can_skip(parts_dict):
         return True
     if 'testing' in parts_dict["activity"]:
         return True
-    if (
-        parts_dict["startText"].strip() in BAD_NAMES_FOR_THINGS or
-        parts_dict["endText"].strip() in BAD_NAMES_FOR_THINGS
+    if not (
+        valid_concept_name(parts_dict["startText"]) or
```
Comment (Joshua-Chin) [inline conceptnet5/readers/conceptnet4.py]: This probably should be an "and"

File: conceptnet5/uri.py [hunk 5]
```diff
@@ -181,7 +100,7 @@ def concept_uri(lang, text, pos=None, disambiguation=None):
         ...
     AssertionError: 'this is wrong' is not in normalized form
     """
-    assert text == standardize_text(text), "%r is not in normalized form" % text
+    assert ' ' not in text, "%r is not in normalized form" % text
```
Comment (Joshua-Chin) [inline conceptnet5/uri.py]: Maybe factor this out into an `is_standardized` function.

File: conceptnet5/readers/conceptnet4.py [hunk 6]
```diff
@@ -85,9 +87,9 @@ def can_skip(parts_dict):
...
+        valid_concept_name(parts_dict["endText"])
     ):
```
Comment (Joshua-Chin) [inline conceptnet5/readers/conceptnet4.py]: This probably should be "and".

File: conceptnet5/uri.py (hunk 5 above)
Comment (rspeer) [inline conceptnet5/uri.py]: It's just a sanity check. I'm saving time by not re-running the actual standardization, just assuming that if you're not standardizing the text, at some point you'll pass in something with a space in it.

File: conceptnet5/nodes.py (hunk 2 above)
Comment (rspeer) [inline conceptnet5/nodes.py]: I'll fix the documentation, but I still want to allow for the possibility of a filter besides `english_filter` later.

**Pull request merged**